File: src/main/java/net/ndrei/teslacorelib/inventory/IFilteredItemHandler.java
package net.ndrei.teslacorelib.inventory;

import net.minecraft.item.ItemStack;
import net.minecraftforge.items.IItemHandlerModifiable;

/**
 * Created by CF on 2016-12-15.
 */
public interface IFilteredItemHandler extends IItemHandlerModifiable {
    boolean canInsertItem(int slot, ItemStack stack);
    boolean canExtractItem(int slot);
}

File: src/main/java/net/ndrei/teslacorelib/inventory/IFilteredItemHandler.kt
package net.ndrei.teslacorelib.inventory

import net.minecraft.item.ItemStack
import net.minecraftforge.items.IItemHandlerModifiable

/**
 * Created by CF on 2017-06-28.
 */
interface IFilteredItemHandler : IItemHandlerModifiable {
    fun canInsertItem(slot: Int, stack: ItemStack): Boolean
    fun canExtractItem(slot: Int): Boolean
}
